Chemistry Recipe fixes (goodbye, concentrated acid!)

Zukulini opened the pull request:
#### Summary

SUMMARY: Content "A series of edits to recipes related to the production of explosive weapons, mainly dynamite, and the products associated."

#### Purpose of change

The game has a bunch of innaccuracies when it comes to the production of dynamite. First, the recipe takes fat directly, instead of glycerol, which is a byproduct of soap and biodiesel production, but that cannot be obtained in the game by any method other than by finding it on specific buildings. On the same topic, glycerol is useless within the game, and it didn't even have a weigh until recently, so I'm giving it the right purpose. I also took the time to delete one legacy lye powder recipe from the game. 

Second, it takes 2 units of either "concentrated acid", sulfuric acid, or nitric acid. "Concentrated acid" is an old legacy item that can be somehow obtained by boiling salt water; this pull takes down all mentions of concentrated acid, water acid, and weak acid water, and replaces them on all the recipes that require them with the appropiate strong acids. On that topic, producing nitroglycerin takes a mix of 2 parts sulfuric acid and 1 part nitric acid. 

#### Describe the solution

I changed all of the recipes to be more realistic. I ran into a couple of problems, however. ~~The first problem relates to the production of LSD. LSD was made with concentrated acid, which is assumed to be a generic acid, so we can infer that in this case it represents the lysergic acid used in the creation of LSD. Of course, in the game there is no way to obtain the lysergic acid that you would need to create LSD... so I just chose to replace it any strong acid as a placeholder.~~

~~Second problem was with the production of neoprene sheets. The production of neoprene requires chloroprene, which I suppose was represented in the recipe by the concentrated acid. I just replaced it in the recipe with all of the three strong acids, even though it would not be possible to make neoprene that way in real life, just as with LSD. In fact, I don't see how it would be possible with the resources you have in the post-apocalypse... maybe we should get rid of the recipe altogether?~~

As for a tip by I-am-erk, I am obsoleting the recipes for LSD and neoprene sheets. 

#### Testing

I tested all of the recipes on my own game and they seemed to work properly. 

For the smoke bomb recipe, given that it takes cola or candies, I assume that it is referring to the popular homemade smokebomb recipe that appears on some publications like the anarchist handbook. I really have no idea why the recipe at the moment takes vitamins or aspirins? I can only imagine that someone a long time ago thought that would be a good substitute for saltpeter. In any case, we now do have saltpeter and sugar, so we can use those instead of what presumably were placeholders back then. 
You can see this recipe in action here: https://youtu.be/RcIMuoUcc1s?t=305

I can't cite my sources because I actually had to calculate the weighs in the formula to make soap myself, but I changed the soap recipe knowing that in real life: 885.4 grams of a triglycerid (tallow or oil) + 120 grams of lye = 919.398 grams of soap + 92.094 grams of glycerol. Thus, the new recipe would use 220 grams of a triglycerid (4 units of tallow, or 14 units of oil) + 29.7 grams (11 units) of lye powder (or 1 unit of lye water, which is a lot more than needed but the units don't get any smaller) and result into 200 grams (10 units) of soap and 20 grams (1 unit) of glycerol. The equivalence is not perfect, that should result in 220~ grams of soap and 22.89 grams of glycerol, but I guess we can assume that some of it is lost in the process given that making it in reality wouldn't be as perfect as the chemical equation. 

As for the glycerol as a result of biodiesel production, this source says that for every 100 pounds of biodiesel produced, 10 pounds of glycerol will turn up as a byproduct. https://farm-energy.extension.org/new-uses-for-crude-glycerin-from-biodiesel-production/

Now as for the dynamite recipe. I'm basing this recipe on the one found in "The Chemistry of Powder and Explosives" by Tenney L. Davis (file is attached), on page 195. As I said before, making dynamite involves a mixture of 2 parts sulfuric acid and 1 part nitric acid. 
![imagen](https://user-images.githubusercontent.com/26130816/88601474-646e8f00-d036-11ea-8baa-2cc0ae43a0fa.png)
Also, making nitroglycerin requires 1/10th of the weigh of nitric acid in glycerol. 
![imagen](https://user-images.githubusercontent.com/26130816/88601694-e6f74e80-d036-11ea-8056-2069a8537760.png)
The recipe has 1 unit of nitric acid (which weighs 380 grams), so I'm making it so that it takes 2 units of glycerol (40 grams, really close to 1/10th of the weigh of the nitric acid). 

[Tenney L. Davis - The Chemistry of Powder and Explosives-Angriff Press (1972).pdf](https://github.com/CleverRaven/Cataclysm-DDA/files/4985088/Tenney.L.Davis.-.The.Chemistry.of.Powder.and.Explosives-Angriff.Press.1972.pdf)

wapcaplet commented:
I don't know chemistry, but must say I appreciate your contributions - we need more versimilitude in this area, especially now that chemistry skill is a thing.

moxian commented:
I have two (and a half) concerns (but i don't feel particularly strongly on either):

1) People making soap (think innawoods, or medieval setting) wouldn't think to collect glycerol byproduct.
And it might make for slightly confusing gameplay: I want to make a soap, but now i have this glycerol thingy that i need a container for, and i was not prepared! And now it's spilled all over the floor, or no!
Making glycerol as a byproduct of a higher-level soap-making recipe (with lower soap yield, in a faux-balance attempt maybe?) would maybe be a good idea?

2) your treatment of different acids in case of LSD and neoprene inconsistent. You insist on using sulphuric for LSD but decide than any will do for neoprene . While I don't think it matters that much which acid is used where, the _reasoning_ you use here does not hold.
Saying that "lysergic acid and sulfuric acid are actually really similar" (I don't know if it's actually true) would be one way to justify this, but as it stands the inconsistency is bothersome.

2.5) Might be a good idea to create a `requirements` LIST for sulfuric/nitric/muriatric acids when those can be used interchangeably (either because all you care is the pH, or because we don't have a more specific acid as in case of LSD/neoprene)

moxian commented on data/json/recipes/weapon/explosive.json:
```diff
@@ -334,8 +329,8 @@
-      [ [ "candy", 1 ], [ "cola", 1 ] ],
-      [ [ "vitamins", 10 ], [ "aspirin", 8 ] ],
+      [ [ "sugar", 16 ] ],
+      [ [ "chem_saltpetre", 24 ] ],
```
This is probably a good change, but completely unrelated to the current PR, as worded.

Zukulini replied:
I'll add that to my description.

Zukulini commented:
> 
> 
> I have two (and a half) concerns (but i don't feel particularly strongly on either):
> 
>     1. People making soap (think innawoods, or medieval setting) wouldn't think to collect glycerol byproduct.
>        And it might make for slightly confusing gameplay: I want to make a soap, but now i have this glycerol thingy that i need a container for, and i was not prepared! And now it's spilled all over the floor, or no!
>        Making glycerol as a byproduct of a higher-level soap-making recipe (with lower soap yield, in a faux-balance attempt maybe?) would maybe be a good idea?
> 
>     2. your treatment of different acids in case of LSD and neoprene inconsistent. You insist on using sulphuric for LSD but decide than any will do for neoprene . While I don't think it matters that much which acid is used where, the _reasoning_ you use here does not hold.
>        Saying that "lysergic acid and sulfuric acid are actually really similar" (I don't know if it's actually true) would be one way to justify this, but as it stands the inconsistency is bothersome.
> 
> 
> 2.5) Might be a good idea to create a `requirements` LIST for sulfuric/nitric/muriatric acids when those can be used interchangeably (either because all you care is the pH, or because we don't have a more specific acid as in case of LSD/neoprene)

I guess I might as well make the LSD recipe work with any strong acid then. I also noticed that but only after I had already sent out the PR, and I didn't really want to wait again for the checks to complete.

Fris0uman commented:
You could also obsolete the recipe of LSD and neoprene that make no sense and add the proper recipes in another PR.

Alex-Folts commented:
I think need to add recipy for making nitric acid from `water_acid` and `water_acid_weak`, as essentially it is very diluted nitric acid. Always thought this need to be done.
Also i think we need at least 2 types of each acid: concentrated and diluted, best would be even 3 types, but at least 2 is good too, and add recipes to make concentrated from diluted and vice verse.

kevingranade commented:
We need a migration entry for the items you are removing (acid and water_acid) so that thy don't cause load errors.

Zukulini commented:
@kevingranade I haven't actually deleted them, I just deleted the recipes that make them and replaced them in the recipes that require them (except two, which I obsoleted).

Should I delete them?

I-am-Erk commented on data/json/items/migration.json:
```diff
@@ -1268,6 +1268,21 @@
+  {
+    "id": "water_acid_weak",
+    "type": "MIGRATION",
+    "replace": "chem_sulphuric_acid"
```
```suggestion
    "replace": "chem_acetic_acid"
```
May as well keep it a relatively weak acid

kevingranade commented on src/weather.cpp:
```diff
@@ -227,7 +225,7 @@ void item::add_rain_to_container( bool acid, int charges )
         return;
     }
-    item ret( acid ? "water_acid" : "water", calendar::turn );
+    item ret( acid ? "water" : "water", calendar::turn );
```
```suggestion
    item ret( "water", calendar::turn );
```
